src/_static/js/workflow.js

```diff
@@ -0,0 +1,71 @@
+class WorkflowContainer {
+    static cellCounter = 0;
+
+    static escapeHtml(unsafe)
+    {
+        return unsafe
+             .replace(/&/g, "&amp;")
+             .replace(/</g, "&lt;")
+             .replace(/>/g, "&gt;")
+             .replace(/"/g, "&quot;")
+             .replace(/'/g, "&#039;");
+     }
+
+    static createWorkflowCell(img, path) {
+        const wrap = document.createElement("div");
+        wrap.classList.add("highlight");
+
+        // workflow SVG renders in a visible pre element
+        const workflowCell = document.createElement("pre");
+        wrap.appendChild(workflowCell);
+        const imgParent = img.parentElement;
+        workflowCell.appendChild(img);
+        imgParent.remove();
+
+        // set button SVG from sphinx-copybutton
+        const id = "workflowcell" + WorkflowContainer.cellCounter++;
+        const clipboardButton = id =>
+        `<button class="copybtn o-tooltip--left" data-tooltip="${messages[locale]['copy']}" data-clipboard-target="#${id}">
+        ${iconCopy}
+        </button>`
+        workflowCell.insertAdjacentHTML('afterend', clipboardButton(id))
+
+        // fetch contents into an invisible pre for copying
+        fetch(path).then(req => req.text()).then(contents => {
+            const codeCell = document.createElement("pre");
+            codeCell.id = id;
+            codeCell.innerHTML = WorkflowContainer.escapeHtml(contents);
+            codeCell.hidden = true;
+            wrap.appendChild(codeCell);
+        });
+        return wrap;
+    }
+
+    static renderElement(element) {
+        element.classList.add("highlight-bonsai");
+        const img = element.querySelector("img");
+
+        // assumes workflow references with $ are hosted in _static/
+        const workflowPath = img.src.replace("$", "_static/");
```

Comment:
I tried this out and `img.src` seems to be a full URL looking like:
http://127.0.0.1:3002/docs/html/reference/reference/$workflows/CentroidModel.bonsai
so only replacing `"$"` will not give us the correct URL.

Wdyt about replacing everything up to `$` with relative URL:
```suggestion
        const workflowPath = img.src.replace(/^.*\$/, "../_static/");
```